src/metadata/selectors.test.js

```diff
@@ -0,0 +1,492 @@
+
+            expect(getCategoryOptionCombos(data)).toBe(expected)
+        })
+    })
+
+    describe('getCategoryOptions', () => {
+        it('returns the expected data', () => {
+            const expected = 'expected'
+            const data = { categoryOptions: expected }
+
+            expect(getCategoryOptions(data)).toBe(expected)
+        })
+    })
+
+    describe('getDataElements', () => {
+        it('returns the expected data', () => {
+            const expected = 'expected'
+            const data = { dataElements: expected }
+
+            expect(getDataElements(data)).toBe(expected)
+        })
+    })
+
+    describe('getDataSets', () => {
+        it('returns the expected data', () => {
+            const expected = 'expected'
+            const data = { dataSets: expected }
+
+            expect(getDataSets(data)).toBe(expected)
+        })
+    })
+
+    describe('getSections', () => {
+        it('returns the expected data', () => {
+            const expected = 'expected'
+            const data = { sections: expected }
+
+            expect(getSections(data)).toBe(expected)
+        })
+    })
+})
+
+describe('simple selectors that select by id', () => {
+    describe('getCategoryById', () => {
+        it('returns the expected data', () => {
+            const id = 'id'
+            const expected = 'expected'
+            const data = {
+                categories: {
+                    [id]: expected,
+                },
+            }
+
+            expect(getCategoryById(data, id)).toBe(expected)
+        })
+    })
+
+    describe('getDataSetById', () => {
+        it('returns the expected data', () => {
+            const id = 'id'
+            const expected = 'expected'
+            const data = {
+                dataSets: {
+                    [id]: expected,
+                },
+            }
+
+            expect(getDataSetById(data, id)).toBe(expected)
+        })
+    })
+
+    describe('getCategoryComboById', () => {
+        it('returns the expected data', () => {
+            const id = 'id'
+            const expected = 'expected'
+            const data = {
+                categoryCombos: {
+                    [id]: expected,
+                },
+            }
+
+            expect(getCategoryComboById(data, id)).toBe(expected)
+        })
+    })
+})
+
+describe('complex selectors that select by id', () => {
+    describe('getSectionById', () => {
+        it('returns the expected data', () => {
+            const dataSetId = 'dataSetId'
+            const sectionId = 'sectionId'
+            const expected = {
+                id: sectionId,
+            }
+            const data = {
+                dataSets: {
+                    [dataSetId]: {
+                        sections: [expected],
+                    },
+                },
+            }
+
+            expect(getSectionById(data, dataSetId, sectionId)).toBe(expected)
+        })
```

Comment:
This is really a non-issue.. Lately I think that it's better to separate the data and the arguments instead of having variables for "shared" ids. I guess the reason why you went for `.toBe` is to ensure that the returned value points to the same reference, but that's not expressed by the test's description, maybe another test case could test for that or the test's description changed to something like `it('returns a reference to the expected data')`, but I don't know if that makes things better or not, maybe it's confusing this way. But as I said, this is a non-issue and more about exchanging thoughts

```suggestion
        it('returns the expected data', () => {
            const data = {
                dataSets: {
                    dataSetId: {
                        sections: [{ id: 'sectionId' }],
                    },
                },
            }
            const expected = { id: 'sectionId' }

            expect(getSectionById(data, 'dataSetId', 'sectionId')).toEqual(expected)
        })
```

Reply:
Hey, thanks for the review. I was talking to Birk about how we wanted to approach this, and we settled on something slightly different. I've rebased the tests on development here: https://github.com/dhis2/data-entry-app/pull/49. We can merge that in, rebase my memoization branch on that, and then after that rebase Birk's branch.

That way we can be sure that behaviour on dev. won't be broken by my memoization changes.

If you don't mind, let me know what you think of: https://github.com/dhis2/data-entry-app/pull/49.